javascript/ql/src/Security/CWE-094/ExpressionInjection.ql

```diff
@@ -78,32 +84,142 @@ private predicate isExternalUserControlledDiscussion(string context) {
+        "\\bgithub\\s*\\.\\s*event\\s*\\.\\s*workflow_run\\s*\\.\\s*display_title\\b",
+        "\\bgithub\\s*\\.\\s*event\\s*\\.\\s*workflow_run\\s*\\.\\s*head_repository\\b\\s*\\.\\s*description\\b",
+        "\\bgithub\\s*\\.\\s*event\\s*\\.\\s*workflow_run\\s*\\.\\s*head_commit\\b\\s*\\.\\s*message\\b",
+        "\\bgithub\\s*\\.\\s*event\\s*\\.\\s*workflow_run\\s*\\.\\s*head_commit\\b\\s*\\.\\s*author\\b\\s*\\.\\s*email\\b",
+        "\\bgithub\\s*\\.\\s*event\\s*\\.\\s*workflow_run\\s*\\.\\s*head_commit\\b\\s*\\.\\s*author\\b\\s*\\.\\s*name\\b",
+        "\\bgithub\\s*\\.\\s*event\\s*\\.\\s*workflow_run\\s*\\.\\s*head_commit\\b\\s*\\.\\s*committer\\b\\s*\\.\\s*email\\b",
+        "\\bgithub\\s*\\.\\s*event\\s*\\.\\s*workflow_run\\s*\\.\\s*head_commit\\b\\s*\\.\\s*committer\\b\\s*\\.\\s*name\\b",
+      ]
+  |
+    context.regexpMatch(reg)
+  )
+}
+
+/**
+ * Holds if the env variable name in `${{ env.name }}`
+ * is where the external user controlled value was assigned to.
+ */
+bindingset[injection]
+predicate isEnvTainted(Actions::Env env, string injection, string context) {
+  Actions::getEnvName(injection) = env.getName() and
+  Actions::getASimpleReferenceExpression(env) = context
+}
+
+/**
+ * Holds if the `run` contains any expression interpolation `${{ e }}`.
+ * Sets `context` to the initial untrusted value assignment in case of `${{ env... }}` interpolation
+ */
+predicate isRunInjectable(Actions::Run run, string injection, string context) {
+  Actions::getASimpleReferenceExpression(run) = injection and
   (
-    exists(on.getNode("issues")) and
-    isExternalUserControlledIssue(context)
-    or
-    exists(on.getNode("pull_request_target")) and
-    isExternalUserControlledPullRequest(context)
-    or
-    (exists(on.getNode("pull_request_review_comment")) or exists(on.getNode("pull_request_review"))) and
-    isExternalUserControlledReview(context)
-    or
-    (exists(on.getNode("issue_comment")) or exists(on.getNode("pull_request_target"))) and
-    isExternalUserControlledComment(context)
+    injection = context
     or
-    exists(on.getNode("gollum")) and
-    isExternalUserControlledGollum(context)
-    or
-    exists(on.getNode("pull_request_target")) and
-    isExternalUserControlledCommit(context)
-    or
-    (exists(on.getNode("discussion")) or exists(on.getNode("discussion_comment"))) and
-    isExternalUserControlledDiscussion(context)
+    isEnvTainted(_, injection, context)
+  )
+}
+
+/**
+ * Holds if the `actions/github-script` contains any expression interpolation `${{ e }}`.
+ * Sets `context` to the initial untrusted value assignment in case of `${{ env... }}` interpolation
+ */
+predicate isScriptInjectable(Actions::Script script, string injection, string context) {
+  exists(Actions::Step step, Actions::Uses uses |
+    script.getWith().getStep() = step and
+    uses.getStep() = step and
+    uses.getGitHubRepository() = "actions/github-script" and
```

Comment:
Isn't this line already covered by the `Script` class? If not, I recommend changing `Script` into a `ScriptStep` class that describes the `actions/github-script` step, which in turn exposes the script text as a field/member predicate.